src/torchmetrics/functional/regression/kendall.py

```diff
@@ -0,0 +1,396 @@
+            f" and {preds.shape[1]}."
+        )
+    if num_outputs == 1:
+        preds = preds.unsqueeze(1)
+        target = target.unsqueeze(1)
+
+    concat_preds.append(preds)
+    concat_target.append(target)
+
+    return concat_preds, concat_target
+
+
+def _kendall_corrcoef_compute(
+    preds: Tensor,
+    target: Tensor,
+    variant: _MetricVariant,
+    alternative: Optional[_TestAlternative] = None,
+) -> Tuple[Tensor, Optional[Tensor]]:
+    """Compute Kendall rank correlation coefficient, and optionally p-value of corresponding statistical test.
+
+    Args:
+        Args:
+        preds: Ordered sequence of data
+        target: Ordered sequence of data
+        variant: Indication of which variant of Kendall's tau to be used
+        alternative: Alternative hypothesis for for t-test. Possible values:
+            - 'two-sided': the rank correlation is nonzero
+            - 'less': the rank correlation is negative (less than zero)
+            - 'greater':  the rank correlation is positive (greater than zero)
+    """
+    (
+        concordant_pairs,
+        discordant_pairs,
+        preds_ties,
+        preds_ties_p1,
+        preds_ties_p2,
+        target_ties,
+        target_ties_p1,
+        target_ties_p2,
+        n_total,
+    ) = _get_metric_metadata(preds, target, variant)
+    con_min_dis_pairs = concordant_pairs - discordant_pairs
+
+    tau = _calculate_tau(
+        preds, target, concordant_pairs, discordant_pairs, con_min_dis_pairs, n_total, preds_ties, target_ties, variant
+    )
+    p_value = (
+        _calculate_p_value(
+            con_min_dis_pairs,
+            n_total,
+            preds_ties,
+            preds_ties_p1,
+            preds_ties_p2,
+            target_ties,
+            target_ties_p1,
+            target_ties_p2,
+            variant,
+            alternative,
+        )
+        if alternative
+        else None
+    )
+
+    # Squeeze tensor if num_outputs=1
+    if tau.shape[0] == 1:
+        tau = tau.squeeze()
+        p_value = p_value.squeeze() if p_value is not None else None
+
+    return tau.clamp(-1, 1), p_value
+
+
+def kendall_rank_corrcoef(
+    preds: Tensor,
+    target: Tensor,
+    variant: Literal["a", "b", "c"] = "b",
+    t_test: bool = False,
+    alternative: Optional[Literal["two-sided", "less", "greater"]] = "two-sided",
+) -> Union[Tensor, Tuple[Tensor, Tensor]]:
+    r"""Computes `Kendall Rank Correlation Coefficient`_.
+
+    .. math::
+        tau_b = \frac{C - D}{\sqrt{(C + D + T_{preds}) * (C + D + T_{target})}}
+
+        tau_c = 2 * \frac{C - D}{n ** 2 * \frac{m - 1}{m}}
+
+    where :math:`C` is represents concordant pairs, :math:`D` stands for discordant pairs and :math:`T` represents
+    a total number of ties. Definition according to `The Treatment of Ties in Ranking Problems`_.
+
+    Args:
+        preds: Ordered sequence of data
+        target: Ordered sequence of data
```

Comment:
should they really be ordered sequences of data?
the data in the examples does not seem to be